src/advanced_agent/config/loader.py

```
@@ -0,0 +1,314 @@
+        file_path = Path(file_path)
+        file_path.parent.mkdir(parents=True, exist_ok=True)
+        
+        config_dict = config.to_dict()
+        
+        with open(file_path, 'w', encoding='utf-8') as f:
+            if format.lower() == "yaml":
+                yaml.dump(config_dict, f, default_flow_style=False, allow_unicode=True)
+            elif format.lower() == "json":
+                json.dump(config_dict, f, indent=2, ensure_ascii=False)
+            else:
+                raise ValueError(f"サポートされていない形式: {format}")
```

Ruff (0.12.2)

Avoid specifying long messages outside the exception class

(TRY003)